dulwich/index.py

```diff
@@ -495,6 +495,27 @@ def has_conflicts(self) -> bool:
                 return True
         return False
 
+    def set_merge_conflict(self, base_entry, base_time, this_entry, this_time, other_entry, other_time):
+        path, base_index_entry, this_index_entry, other_index_entry = None, None, None, None
+        if this_entry:
+            path = this_entry.path
+            this_index_entry = IndexEntry(this_time, this_time, 0, 0, this_entry.mode,
+                                          0, 0, 0, this_entry.sha)
+        if base_entry:
+            if not path:
+                path = base_entry.path
+            base_index_entry = IndexEntry(base_time, base_time, 0, 0, base_entry.mode,
+                                          0, 0, 0, base_entry.sha)
+        if other_entry:
+            if not path:
+                path = other_entry.path
+            other_index_entry = IndexEntry(other_time, other_time, 0, 0, other_entry.mode,
+                                           0, 0, 0, other_entry.sha)
+        if path:
+            self._byname[path] = ConflictedIndexEntry(ancestor=base_index_entry,
```

Comment:
```suggestion
            self[path] = ConflictedIndexEntry(ancestor=base_index_entry,
```

Reply:
Okay.

FWIW, all of these changes were part of my earlier PR that you accepted and then reworked completely.

I will work on the requested test additions.

Reply:
I intentionally changed the code back to preserve the original API that uses names as keys,  requiring people to explicitly specify which stage they're wanting to deal with for conflicted entries.

Reply:
That  could also be done in the original PR you accepted, and then you changed it almost completely to the point of even dropping the interface needed to set a conflict!  Your changes created all of these issues.  I am just trying to get back to the functionality it had when you first accepted it.

Reply:
The original PR had an ambiguous interface (allowing both names and (name, stage) tuples). It implicitly defaulted to "this" for anything that was conflicted.

Reply:
How on earth is allowing a key that is a path or tuple or any more ambiguous than allowing a value to be either of two very different classes?   And then breaking things and dropping the interface?

After 40+ years of being a paid and opensource developer who has made contributions to many many projects, I can see this is just not going to work.  You place more hurdles to getting working code going than anyone I have ever met.  Please close this.

Sigil will fork and go our own way.

Reply:
Sorry :-( That was not my intent; I made that change specifically so that at least the PR itself could land, and perhaps that was the wrong call.

Maybe to give a expand on that justification: I feel that Index with current implementation is more pythonic. You could replace "Index" with "dict", and it would mostly work. It's not uncommon for different types to be used for the values of a dict, but seems far more surprising that changing one key deletes another.